pkg/sriov/sriov.go

```diff
@@ -245,6 +252,9 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 	}
 
 	// 2. Set mac address
+	if err = s.nLink.LinkSetVfHardwareAddr(pfLink, conf.VFID, *conf.MAC); err != nil {
```

Comment:
On i40e the VF interface will ignore it and keep using the old MAC inside the pod. I also found that after you bind VF to the userspace driver and then bind it back to the i40evf, it picks up the new MAC configured on the PF.

IIRC in the DPDK applications it will pick up the new MAC correctly during EAL initialization so that wouldn't be a problem there - this needs to be verified anyway.

Reply:
A side question, how the dpdk application get to know the MAC address inside pod? by looking at the `proc`?

Reply:
It uses one that is stored in the PF configuration or if there's none it generates a random one - [that](https://git.dpdk.org/dpdk/tree/drivers/net/i40e/i40e_ethdev_vf.c#n1276) would be the code if you're really curious :) Of course the implementation may differ for other vendors' NICs.

Reply:
This whole business of MAC registration for VFs seems to be full of vendor-driver specific kludges. I wonder if we should mess with unbind/bind to apply new MAC address to consistently VF netlink devices, or consider it a bug in the kernel driver...

Reply:
For example here with igbvf driver, when using the latest version of this PR, I see MAC address applied to netlink device:

```
[root@hera05 ~]# kubectl exec -it  -n kubevirt-hyperconverged   testpod1 ip link              
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
3: eth0@if1418: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP mode DEFAULT group default 
    link/ether de:0c:57:58:bb:d7 brd ff:ff:ff:ff:ff:ff link-netnsid 0
1376: net1: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN mode DEFAULT group default qlen 1000
    link/ether 02:00:00:ff:ee:dd brd ff:ff:ff:ff:ff:ff
```

Though reset of the same doesn't seem to take into effect, so even when I delete the pod, I still have the VF carrying the overridden MAC address:

```
[root@hera05 ~]# ip link | grep 02:00:00:ff:ee:dd         
    vf 22 MAC 02:00:00:ff:ee:dd, spoof checking on, link-state auto, trust off, query_rss off
    vf 51 MAC 02:00:00:ff:ee:dd, spoof checking on, link-state auto, trust off, query_rss off
    link/ether 02:00:00:ff:ee:dd brd ff:ff:ff:ff:ff:ff
    link/ether 02:00:00:ff:ee:dd brd ff:ff:ff:ff:ff:ff
```

Also in dmesg I see:

```
[23939.924656] ixgbe 0000:05:00.1 em4: VF 22 attempted to override administratively set MAC address
Reload the VF driver to resume operations
```

All these issues don't seem to affect DPDK mode though because there we don't have any VF netlink drivers to deal with.